soh/src/code/z_play.c

```diff
@@ -929,7 +931,11 @@ void Gameplay_Update(GlobalContext* globalCtx) {
                     LOG_NUM("1", 1);
                 }
 
-                globalCtx->gameplayFrames++;
+                if (gSaveContext.fileNum == 0xFF) {
+                    globalCtx->gameplayFrames = 0;
```

Comment:
Is the intention here to prevent autosave on the debug save? If so I don't think we want to achieve that like this, lots of things watch gameplayFrames

Reply:
No, this has nothing to do with debug saves, other than the fact that they're also affected by gameplayFrames never being initialized and always using garbage data.  The intention is to always start gameplayFrames at 0 when you first load a file.  That way, autosave can be prevented during the first few seconds, since that was causing the issue linked above where your magic was empty when loading a file (technically it was starting at 0 and animating upward, but the save triggered before the animation could start).  

However, if you think keeping gameplayFrames at 0 whenever you're using the debug file will cause other problems, I can find another place to initialize it.  In fact, now that we have #1854 in the works, hopefully this will give a single canonical place to initialize all file-dependent variables and will fully resolve the above issue, which was still occurring when using fast file select.  Either way, we still absolutely need to set gameplayFrames to 0 _somewhere_ since using uninitialized variables is not good.

Reply:
My initial impression was that this was going to run every frame, keeping gameplayFrames at 0 forever for debug saves. If this is not the case you can disregard and this is good to go, otherwise let’s move it to somewhere that’s only run once.

Reply:
You're correct, it does keep it at 0, so I'll find a different place for it.

Reply:
Yup, I plan on doing it tomorrow night; I was waiting on #1927 and am just giving it a couple more days to make sure nothing else big comes in too 😉